### PR TITLE
Try get async

### PR DESCRIPTION
**Pull request by mumby0168**

### CODE REVIEW

File: src/Microsoft.Azure.CosmosRepository/Repositories/DefaultRepository.Read.cs [hunk 1]
```diff
@@ -5,15 +5,30 @@
...
 using Microsoft.Azure.Cosmos;
 using Microsoft.Azure.CosmosRepository.Logging;
 
+// ReSharper disable once CheckNamespace
 namespace Microsoft.Azure.CosmosRepository
 {
-    internal sealed partial class DefaultRepository<TItem>
+    internal sealed partial class DefaultRepository<TItem> where TItem : class, IItem
     {
+        public async ValueTask<TItem?> TryGetAsync(string id, string? partitionKeyValue = null, CancellationToken cancellationToken = default)
```
Comment (IEvangelist) [inline src/Microsoft.Azure.CosmosRepository/Repositories/DefaultRepository.Read.cs:19]:
```suggestion
        public async ValueTask<TItem?> TryGetAsync(
            string id,
            string? partitionKeyValue = null,
            CancellationToken cancellationToken = default)

```

File: src/Microsoft.Azure.CosmosRepository/Repositories/DefaultRepository.Read.cs [hunk 2]
```diff
@@ -5,15 +5,30 @@
...
+        {
+            try
+            {
+                return await GetAsync(id, partitionKeyValue, cancellationToken);
+            }
+            catch (CosmosException e) when (e.StatusCode is HttpStatusCode.NotFound)
+            {
+                _logger.LogItemNotFoundHandled<TItem>(id, partitionKeyValue ?? id, e);
+                return null;
```
Comment (IEvangelist) [inline src/Microsoft.Azure.CosmosRepository/Repositories/DefaultRepository.Read.cs:28]:
```suggestion
                return default;
```

If you use the `default` keyword here instead, you do not need the generic-type `class` constraint.

File: src/Microsoft.Azure.CosmosRepository/Repositories/InMemoryRepository.cs [hunk 3]
```diff
@@ -61,6 +62,18 @@ internal class InMemoryRepository<TItem> : IRepository<TItem> where TItem : clas
         internal TDeserializeTo DeserializeItem<TDeserializeTo>(string jsonItem) =>
             JsonConvert.DeserializeObject<TDeserializeTo>(jsonItem);
 
+        public async ValueTask<TItem?> TryGetAsync(string id, string? partitionKeyValue = null, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                return await GetAsync(id, partitionKeyValue, cancellationToken);
+            }
+            catch (CosmosException e) when (e.StatusCode is HttpStatusCode.NotFound)
```
Comment (IEvangelist) [inline src/Microsoft.Azure.CosmosRepository/Repositories/InMemoryRepository.cs:71]: Are we internally (within the delegated `GetAsync`) throwing a `CosmosException`? If not, we'd probably want to alter this a bit, no?

File: src/Microsoft.Azure.CosmosRepository/Repositories/InMemoryRepository.cs [hunk 4]
```diff
@@ -61,6 +62,18 @@ internal class InMemoryRepository<TItem> : IRepository<TItem> where TItem : clas
...
+            {
+                return null;
```
Comment (IEvangelist) [inline src/Microsoft.Azure.CosmosRepository/Repositories/InMemoryRepository.cs:73]:
```suggestion
                return default;
```

File: src/Microsoft.Azure.CosmosRepository/Repositories/DefaultRepository.cs [hunk 5]
```diff
@@ -1,97 +1,98 @@
...
+using Microsoft.Azure.CosmosRepository.Specification.Evaluator;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+// ReSharper disable once CheckNamespace
+namespace Microsoft.Azure.CosmosRepository
+{
+    /// <inheritdoc/>
+    internal sealed partial class DefaultRepository<TItem> : IRepository<TItem>
+        where TItem : class, IItem
```
Comment (IEvangelist) [inline src/Microsoft.Azure.CosmosRepository/Repositories/DefaultRepository.cs:23]: I believe that we should be able to drop the `class` constraint here. Also, we only technically need one implementation of the `partial` to have the generic type constraints. That is why I only had it on this core implementation here (I'm ok with having it on both though, if you prefer that).

**Review by IEvangelist — APPROVED** [review]
A few nits, and suggestions.

File: src/Microsoft.Azure.CosmosRepository/Repositories/InMemoryRepository.cs (hunk 3 above)
Comment (mumby0168) [reply src/Microsoft.Azure.CosmosRepository/Repositories/InMemoryRepository.cs:71]: We are throwing that in there, the method `NotFound()` throws one.

File: src/Microsoft.Azure.CosmosRepository/Repositories/InMemoryRepository.cs (hunk 4 above)
Comment (mumby0168) [reply src/Microsoft.Azure.CosmosRepository/Repositories/InMemoryRepository.cs:73]: This is cool I did not know this!

File: src/Microsoft.Azure.CosmosRepository/Repositories/DefaultRepository.cs (hunk 5 above)
Comment (mumby0168) [reply src/Microsoft.Azure.CosmosRepository/Repositories/DefaultRepository.cs:23]: Yeah I did think that, I think I did a bad merge. Updated.

**Review by IEvangelist — APPROVED** [review]
Looks good to me, :shipit: when you're ready.

**Pull request merged**